WordPress/Classes/Services/HomepageSettingsService.swift

```diff
@@ -10,69 +10,81 @@ struct HomepageSettingsService {
+    init?(blog: Blog, coreDataStack: CoreDataStack) {
         guard let api = blog.wordPressComRestApi(), let dotComID = blog.dotComID as? Int else {
             return nil
         }
 
         self.remote = HomepageSettingsServiceRemote(wordPressComRestApi: api)
         self.siteID = dotComID
         self.blog = blog
-        self.context = context
+        self.coreDataStack = coreDataStack
     }
 
     public func setHomepageType(_ type: HomepageType,
                                 withPostsPageID postsPageID: Int? = nil,
                                 homePageID: Int? = nil,
                                 success: @escaping () -> Void,
                                 failure: @escaping (Error) -> Void) {
+        var originalHomepageType: HomepageType?
+        var originalHomePageID: Int?
+        var originalPostsPageID: Int?
+        coreDataStack.performAndSave { context in
+            guard let blog = Blog.lookup(withObjectID: self.blog.objectID, in: context) else {
+                return
+            }
 
-        // Keep track of the original settings in case we need to revert
-        let originalHomepageType = blog.homepageType
-        let originalHomePageID = blog.homepagePageID
-        let originalPostsPageID = blog.homepagePostsPageID
+            // Keep track of the original settings in case we need to revert
+            originalHomepageType = blog.homepageType
+            originalHomePageID = blog.homepagePageID
+            originalPostsPageID = blog.homepagePostsPageID
 
-        switch type {
-        case .page:
-            blog.homepageType = .page
-            if let postsPageID = postsPageID {
-                blog.homepagePostsPageID = postsPageID
-                if postsPageID == originalHomePageID {
-                    // Don't allow the same page to be set for both values
-                    blog.homepagePageID = 0
+            switch type {
+            case .page:
+                blog.homepageType = .page
+                if let postsPageID = postsPageID {
+                    blog.homepagePostsPageID = postsPageID
+                    if postsPageID == originalHomePageID {
+                        // Don't allow the same page to be set for both values
+                        blog.homepagePageID = 0
+                    }
                 }
-            }
-            if let homePageID = homePageID {
-                blog.homepagePageID = homePageID
-                if homePageID == originalPostsPageID {
-                    // Don't allow the same page to be set for both values
-                    blog.homepagePostsPageID = 0
+                if let homePageID = homePageID {
+                    blog.homepagePageID = homePageID
+                    if homePageID == originalPostsPageID {
+                        // Don't allow the same page to be set for both values
+                        blog.homepagePostsPageID = 0
+                    }
                 }
```

Comment:
I mention in my comment above that we _could_ use an early return here. 

Suggestion, just to see how it looks like:

```suggestion
                guard let homePageID else { return }
                
                blog.homepagePageID = homePageID
                
                guard homePageID == originalPostsPageID else { return }
                
                // Don't allow the same page to be set for both values
                blog.homepagePostsPageID = 0
```